cosmetics-web/app/controllers/responsible_persons/notifications_controller.rb

```diff
@@ -14,7 +13,21 @@ def index
     end
   end
 
-  def show; end
+  def archived
+    @registered_notifications = get_registered_archived_notifications(20)
+    respond_to do |format|
+      format.html
+      format.csv do
+        @notifications = NotificationsDecorator.new(@responsible_person.notifications.archived.order(notification_complete_at: :desc))
```

Comment:
Suggestion:

Noticed that the CSV format doesn't use `@registered_notifications` and that we retrieve the notifications twice:
- Once for `@registered_notifications` with pagination.
- Once with the same query but without paginating for the decorated `@notifications`.
[Kaminari offers a way to unscope a paginated query](https://regulatorydelivery.atlassian.net/browse/COSBETA-1861) that looks perfect for this use case


```suggestion
        @notifications = NotificationsDecorator.new(@registered_notifications.except(:limit, :offset))
```